src/Adapter/Form/ChoiceProvider/CountryStateByIdChoiceProvider.php

```diff
@@ -47,14 +48,21 @@ public function getChoices(array $options)
         $resolvedOptions = $resolver->resolve($options);
 
         try {
-            $states = State::getStatesByIdCountry($resolvedOptions['id_country'], $resolvedOptions['only_active']);
+            $countryId = $resolvedOptions['id_country'];
+            $countryHasStates = (new Country($countryId))->contains_states;
+
+            if (!$countryHasStates) {
+                return [];
+            }
+
+            $states = State::getStatesByIdCountry($countryId, $resolvedOptions['only_active']);
             $choices = [];
```

Comment:
`$choices` must be decalred above. If there is an exception in `new Country`, `$choices` will be undefined

Reply:
I get what you mean it would be cleaner, but it shoudn't be a problem here since an exception is thrown in the catch block It will never reach the `return $choices;`

Reply:
What if it's not a `PrestaShopException` :thinking:

Reply:
Then it's not caught by this class and either the process crashes or it's caught by something higher In both cases it still won't reach the `return`

Reply:
Could you just move the `return $choices` after the foreach to have a cleaner code :D :pray: